i18n/src/main/scala/io/udash/i18n/TranslationKey.scala

```diff
@@ -97,12 +100,23 @@ object TranslationKey {
     new Untranslatable(key)
 
   private[i18n] final class ReducedTranslationKey(key: String, val argv: Any*) extends TranslationKey0(key) {
+    override def productPrefix: String = "ReducedTranslationKey"
+    override def productArity: Int = 2
+    override def productElement(n: Int): Any = n match {
+      case 0 => key
+      case 1 => argv
+      case _ => throw new IndexOutOfBoundsException
+    }
+
+    override def toString(): String = s"$productPrefix($key,${argv.mkString(",")})"
+
     override def apply()(implicit provider: TranslationProvider, lang: Lang): Future[Translated] = {
       provider.translate(key, argv: _*)
     }
   }
 
   private[i18n] final class Untranslatable(key: String) extends TranslationKey0(key) {
+    override def productPrefix: String = "Untranslatable"
     override def apply()(implicit provider: TranslationProvider, lang: Lang): Future[Translated] =
       Future.successful(Translated(key))
   }
```

Comment:
TBH I'm not in love with the design (before your changes as well). Maybe we could break things up a little more, make `TranslationKey0` a trait / abstract class and avoid case class overrides? I don't think the `TranslationKey0` constructor is used very often (there are factories) and we could still provide the `apply` for compatibility. WDYT?

Reply:
You mean splitting `TranslationKey0` into a trait (from which `ReducedTranslationKey` and `Untranslatable` would inherit) and a case class? Good idea, but now the hard part is: how to name them?